src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Services/Implementation/ContextHandler.cs

```diff
@@ -297,6 +300,54 @@ protected async Task<List<Role>> GetRoles(int subjectUserId, int resourcePartyId
             return roles;
         }
 
+        /// <summary>
+        /// Gets the list of mainunits for a subunit
+        /// </summary>
+        /// <param name="mainUnitQuery">Query model for the list of subunit partyIds to check and retrieve mainunits for</param>
+        /// <returns>List of mainunits</returns>
+        protected async Task<List<MainUnit>> GetMainUnits(MainUnitQuery mainUnitQuery)
+        {
+            string cacheKey = $"GetMainUnitsFor:{string.Join(",", mainUnitQuery.PartyIds)}";
```

Comment:
This appears to always be called with a single PartyID, so this is probably just academic, but ideally this should cache individual partyids and only call Bridge with those ids not currently cached. Or instead just append `.Sort()` to avoid order of ids to have any effect.

Reply:
Correct, it's only used with a single subunit as input. I was unsure how critical it was to get the patch out, so didn't want to spend time needlessly optimising.

I've instead now changed the implementation to only take the single subunit partyid as input.

If a future change needs that functionality it might not need or want caching anyway,
and it shouldn't use DelegationContextHandler anyway but either use _partiesWrapper directly or through it's own Handler implementation.